app/models/goal.py

```diff
@@ -3,3 +3,23 @@
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+
+    @classmethod
+    def from_dict(cls, data_dict):
+        return cls(
+                    title=data_dict["title"]
+        )
+
+    
+    def to_dict(self):
+        return dict(
+                    id=self.goal_id,
+                    title=self.title
+        )
```

Comment:
The indentation inside the returns in the `from_dict` and `to_dict` functions looks deeper than expected, I recommend removing the extra indentation to be consistent with whitespace across functions in the project:
```suggestion
        return dict(
            id=self.goal_id,
            title=self.title
        )
```